src/Common/src/CoreLib/System/ReadOnlyMemory.cs

```diff
@@ -168,7 +168,11 @@ public ReadOnlySpan<T> Span
                 }
                 else if (typeof(T) == typeof(char) && _object is string s)
                 {
+#if CORECLR || CORERT
```

Comment:
We should have a different ifdef for this so that this is just one condition.

It can be `#if netstandard` or `#if FEATURE_PORTABLE_SPAN`. We have prior art for both these.